java-security/src/main/java/com/sap/cloud/security/token/AbstractToken.java

```diff
@@ -39,6 +39,7 @@
 	protected final DefaultJsonObject tokenBody;
 
 	public AbstractToken(@Nonnull DecodedJwt decodedJwt) {
+		LOGGER.debug("Decoded Jwt: HEADER - {}, PAYLOAD - {}, SIGNATURE - {}", decodedJwt.getHeader(), decodedJwt.getPayload(), decodedJwt.getSignature());
```

Comment:
there is already a similar implementation available in Base64JwtDecoder.DecodedJwtImpl.toString() - maybe it can be reused / enhanced?